webpack.common.js

```diff
@@ -0,0 +1,121 @@
+// @flow
```

Comment:
It seems it's possible to
(i) use ES6 in `webpack.config.js` by using the extension `.babel.js`, which transpiles the file before loading
(ii) put all configurations in one file by `webpack.config.js` exporting a function, which takes environment variables as an argument and returns the webpack configuration

Reply:
I think having the whole config in a single file and using environment variables to control things might be nice.  I'm going land this as is though and any additional work could happen in a followup PR.